tests/functional/filter/cram/subsample-weighted-and-uniform-mix.t

```diff
@@ -0,0 +1,109 @@
+Setup
+
+  $ source "$TESTDIR"/_setup.sh
+
```

Comment:
If we add a test which produces 50 x A and 150 x B, like so:

```
  $ echo "strain	date	location" > metadata.tsv
  $ for i in $(seq 1 50); do
  >     echo "2000A_$i	2000	A" >> metadata.tsv
  >     echo "2000B_$i	2000	B" >> metadata.tsv
  >     echo "2001B_$i	2001	B" >> metadata.tsv
  >     echo "2002B_$i	2002	B" >> metadata.tsv
  > done
```

and then ask for a 1:1 ratio of A:B with `--subsample-max-sequences 200` then we target 100xA, 100xB which isn't possible. Not a whole lot we can do here I don't think, but we should at least add a note indicating that certain groups (e.g. "A") didn't have enough samples and thus the output distribution doesn't match the intended weights. 

(This behaviour holds across both `--group-by location` and `--group-by location year`, although the latter can cause extra complications -- see the other review thread.)

Reply:
The solution here is the same as in the other thread, see my comment:

> The missing piece there (and also what's needed to produce the additional output in your suggestion) is to update this function to be aware of the number of rows per group available in the metadata. It should be doable, I will take a look.

Reply:
Addressed with warning in output. Example:

```
WARNING: Targeted 3 sequences for group ["country='A'"] but only 1 is available.
```